backend/app/controllers/menu-item.controller.js

```diff
@@ -0,0 +1,59 @@
+const config = require("../config/auth.config");
+const db = require("../models");
+const MenuItem = db.menuitem;
+const Category = require("../models/category.model");
+
+exports.addMenuItem = (req, res) => {
+  const menuitem = new MenuItem({
+    meal: req.body.meal,
+    category: req.body.category,
+  });
+
+  menuitem.save((err, menuitem) => {
+    if(err) {
+      res.status(500).send({ message: err });
+      return;
+    }
+
+    if (req.body.categories) {
```

Comment:
Should this be `req.body.category`?

Reply:
Resolved that as discussed. 

I decided to read more on populate.. https://mongoosejs.com/docs/populate.html

Going by what's there.. looks like I need just the meal and category schemas, and then I can add category to the meal Schema directly. I'll now have:

const menuItem = new Meal({
  _id: new mongoose.Types.ObjectId(),
  name: ' ',
  description: ' ',
  image: ' ',
  price: ' ',
});

menuItem.save(function (err) {
  if (err) return handleError(err);

  const category1 = new Category({
    title: 'Snacks',
    menuItem: menuItem._id    // problem I have now is how do I handle this, since one category can belong to 
                                                 //several menuItems
  });

Reply:
Hold on.. there's a long list on populate. I'm reading about Populate Virtuals now

Reply:
Virtuals are an interesting implementation here.

Reply:
Lool interesting as in "go ahead?" or "why don't you try another approach?"